src/db/gc/mod.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+//!
+//! The current implementation of the garbage collector is concurrent mark-and-sweep.
+//!
+//! ## Design goals
+//! A correct GC algorithm that is simple and efficient for forest scenarios.
+//!
+//! ## GC Algorithm
+//! The `mark-and-sweep` algorithm was chosen due to it's simplicity, efficiency and low memory
+//! footprint. We do have to generate keys from values for some of the iterated columns due to a
+//! ParityDB limitation. See <https://github.com/paritytech/parity-db/issues/187>.
+//! Previously the `semi-space` algorithm was used resulting in data duplication and up to 100%
+//! extra disk usage.
+//!
+//! ## GC Workflow
+//! 1. Mark: traverse all the relevant database columns, generating integer hash representations for
+//! each database key and storing those in a [`HashSet`].
```

Comment:
That they're stored in a `HashSet` is an implementation detail and doesn't have to be mentioned here.

Reply:
`set` is a good enough then.